src/EnrollOnPremController.js

```diff
@@ -95,7 +95,16 @@ function (Okta, FormType, FormController, Footer, TextBox) {
       };
     },
 
-    Footer: Footer
+    Footer: Footer,
+
+    initialize: function () {
+      var factors = this.options.appState.get('factors');
+      var factor = factors.findWhere(_.pick(this.options, 'provider', 'factorType'));
```

Comment:
I think this might be simpler if it is added in the Model, around lines 36-44, using the default value for the model prop. Something like
```
var provider = this.options.provider;
var credentiaId = undefined // Get the default value here
return {
  props: {
    credentialId: ['string', true, credentialId],
    passCode: ['string', true],
    factorId: 'string'
  },
```